src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -351,6 +351,21 @@ public static String recoverFromSanitizedText(String str) {
     }
     
     /**
+     * Recovers text that has been sanitized for javascript under Sanitizer.sanitizeForJs()
+     */
+    public static String recoverFromSanitizedForJsText(String str) {  
+        if (str == null) {
+            return null;
+        }
+        
+        return recoverFromSanitizedText(str)
+                  .replace("\\\\", "\\")
+                  .replace("\\\"", "\"")
+                  .replace("\\'", "'")
+                  .replace("\\#", "#");
```

Comment:
If there is no other special character allowed in session names besides `#` then that should be fine.